week3/homework/src/manipulate.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const fs = require('fs');
+const storage = './todos.json';
+
+function runTheFunction(func, param1, param2) {
+    fs.readFile(storage, 'utf8', (err, data) => {
+        if (err) {
+            if (err.errno === -4058) {
+                fs.appendFile(storage, '[]', (error) => {
+                    if (error) { console.log(error); }
+
+                    fs.readFile(storage, 'utf8', (error, data) => {
+                        if (error) {
+                            console.log(error);
+                        } else {
+                            allFunctions(func, data, param1, param2);
+                        }
+                    });
+                });
+            } else {
+                console.log(err);
+            }
+        } else {
+            allFunctions(func, data, param1, param2);
+        }
+    });
+}
+
+function allFunctions(func, data, param1, param2) {
+    switch (func) {
+        case 'add':
+            add(data, param1);
+            break;
+        case 'mark':
+            mark(data, param1, param2);
+            break;
+        case 'list':
+            list(data, param1, param2);
+            break;
+    }
+}
+
+function add(data, item) {
+    const toDoList = JSON.parse(data);
+    const itemObj = {
+        index: toDoList.length + 1,
+        description: item
+    };
+    toDoList.push(itemObj);
+
+    fs.writeFile(storage, JSON.stringify(toDoList), (error) => {
+        if (error) { console.log(error); }
+    })
+}
+
+function mark(data, index, type) {
+    const toDoList = JSON.parse(data);
+    toDoList[index - 1].done = type;
+    fs.writeFile(storage, JSON.stringify(toDoList), (error) => {
```

Comment:
This could also be a seperate function `writeData(toDoList)`, since you are using it in different places.